src/types/params/plain_secret.rs

```diff
@@ -328,6 +360,47 @@ impl PlainSecretParams {
                         data
                     }
                     KeyVersion::V5 => unimplemented_err!("v5 encryption"),
+                    KeyVersion::V6 => unimplemented_err!("v6 encryption"),
+                    KeyVersion::Other(v) => unimplemented_err!("encryption for key version {}", v),
+                };
+
+                Ok(EncryptedSecretParams::new(enc_data, s2k_params))
+            }
+            S2kParams::Aead {
+                sym_alg,
+                aead_mode,
+                s2k,
+                nonce,
+            } => {
+                let key = s2k.derive_key(passphrase, sym_alg.key_size())?;
+
+                let enc_data = match version {
+                    KeyVersion::V2 => unsupported_err!("Encryption for V2 keys is not available"),
+                    KeyVersion::V3 => unimplemented_err!("v3 encryption"),
+                    KeyVersion::V4 => unimplemented_err!("v4 aead encryption"), // FIXME: implement
```

Comment:
don't we already have aead? or is this for the next round ?

Reply:
key encryption with aead will be handled in the followup PR